template.yaml

```diff
@@ -632,6 +636,423 @@ Resources:
+      AlarmDescription: Trigger the alarm if less than 99% of requests has a latency of less than 2.5 second with a minimum of 150 invocations in 2 out of the last 5 minutes
+      ActionsEnabled: false
+      OKActions:
+        - !ImportValue platform-alarm-topic-slack-warning-alert
+      AlarmActions:
+        - !ImportValue platform-alarm-topic-slack-warning-alert
+      InsufficientDataActions: [ ]
+      Dimensions: [ ]
+      EvaluationPeriods: 5
+      DatapointsToAlarm: 2
+      Threshold: 2500
+      ComparisonOperator: GreaterThanOrEqualToThreshold
+      TreatMissingData: notBreaching
+      Metrics:
+        - Id: latencyThreshold
+          Label: latencyThreshold
+          ReturnData: true
+          Expression: IF(invocations<150,0,latency99Percentile)
+        - Id: invocations
+          ReturnData: false
+          MetricStat:
+            Metric:
+              Namespace: AWS/ApiGateway
+              MetricName: Count
+              Dimensions:
+                - Name: ApiId
+                  Value: !Ref ApiGwHttpEndpoint
+            Period: 60
+            Stat: Sum
+        - Id: latency99Percentile
+          ReturnData: false
+          MetricStat:
+            Metric:
+              Namespace: AWS/ApiGateway
+              MetricName: Latency
+              Dimensions:
+                - Name: ApiId
+                  Value: !Ref ApiGwHttpEndpoint
+            Period: 60
+            Stat: p99
+
+  FELowContainerTaskCountAlarm:
+    DependsOn:
+      - F2FFrontEcsService
+      - F2FFrontEcsCluster
+    Type: AWS::CloudWatch::Alarm
+    Properties:
+      AlarmName: !Sub "${AWS::StackName}-FELowContainerTaskCountAlarm"
+      AlarmDescription: Trigger a warning if the running container task count drops below 2
+      ActionsEnabled: true
+      AlarmActions:
+        - !ImportValue platform-alarm-topic-critical-alert
+      InsufficientDataActions: []
+      Dimensions: []
+      EvaluationPeriods: 1
+      DatapointsToAlarm: 1
+      Threshold: 1
+      ComparisonOperator: LessThanOrEqualToThreshold
+      TreatMissingData: breaching
+      Metrics:
+        - Id: containerTaskCount
+          ReturnData: true
+          MetricStat:
+            Metric:
+              Namespace: ECS/ContainerInsights
+              MetricName: RunningTaskCount
+              Dimensions:
+                - Name: ClusterName
+                  Value: !Ref F2FFrontEcsCluster
+                - Name: ServiceName
+                  Value: !GetAtt F2FFrontEcsService.Name
+            Period: 60
+            Stat: Minimum
+
+  FEHighContainerTaskCountWarning:
+    DependsOn:
+      - F2FFrontEcsService
+      - F2FFrontEcsCluster
+    Type: AWS::CloudWatch::Alarm
+    Properties:
+      AlarmName: !Sub "${AWS::StackName}-FEHighContainerTaskCountWarning"
+      AlarmDescription: >-
+        Trigger a warning if the running container task count reaches the auto-scaling maximum for 5 minutes
```

Comment:
not sure if this is ever going to happen should we have lower limit.

Reply:
yes we should have a low limit alarm. If the container counts is <3 then warning, <2 then critical. Reason being that if we get down to just 1 container it is only going to be in 1 AZ, hence a single point of failure

Reply:
Create a separate ticket to discuss the details.